apps/root/src/pages/earn/components/all-strategies-table/components/toolbar/index.tsx

```diff
@@ -1,21 +1,53 @@
 import React from 'react';
 import { useThemeMode } from '@state/config/hooks';
-import { FormattedMessage } from 'react-intl';
-import { ContainerBox, Typography, colors } from 'ui-library';
+import { FormattedMessage, defineMessage, useIntl } from 'react-intl';
+import { ContainerBox, InputAdornment, SearchIcon, TextField, Typography, colors } from 'ui-library';
 import TableFilters from '../filters';
 
 interface AllStrategiesTableToolbarProps {
   isLoading: boolean;
+  search: string;
+  setSearch: (newValue: string) => void;
 }
 
-const AllStrategiesTableToolbar = ({ isLoading }: AllStrategiesTableToolbarProps) => {
+const AllStrategiesTableToolbar = ({ isLoading, search, setSearch }: AllStrategiesTableToolbarProps) => {
+  const intl = useIntl();
   const themeMode = useThemeMode();
   return (
     <ContainerBox justifyContent="space-between" alignItems="end">
       <Typography variant="h4Bold" color={colors[themeMode].typography.typo1}>
         <FormattedMessage description="earn.all-strategies-table.title" defaultMessage="All Vaults" />
       </Typography>
-      <TableFilters isLoading={isLoading} />
+      <ContainerBox gap={6}>
+        <TextField
+          size="small"
+          placeholder={intl.formatMessage(
+            defineMessage({
+              defaultMessage: 'Search by Vault, Chain, Assets, Guardian or Yield Type',
+              description: 'allStrategiesSearch',
+            })
+          )}
+          value={search}
+          onChange={(evt: React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>) =>
+            setSearch(evt.currentTarget.value)
```

Comment:
I'm thinking if we add this to the redux state, we wont need to:
1. Filter strategies on the parent component
2. Reduce prop drilling for search parameters.

What do you think?

Reply:
Like it very much sir!